app/main/lib/openai.py

```diff
@@ -0,0 +1,28 @@
+from flask import current_app as app
+import openai.embeddings_utils
+import redis
+import hashlib
+import pickle
+
+PREFIX_OPENAI = "openai-"
+EMBEDDING_CACHE_DEFAULT = 60*60*24*7 #7 days (value is in seconds)
+def retrieve_openai_embeddings(text, model_key):
+
```

Comment:
Might be nice to move the cache-specific operations into separate function so that part can be exercised/tested independently from the retrieve embeddings funciton.

Reply:
```
$ echo "Waiting for model servers..." && while [[ ! '2' =~ $(redis-cli -n 1 SCARD 'SharedModel') ]]; do sleep 1; done

Waiting for model servers...
```
looks like build died, maybe isn't fulling bringing up redis in the CI?

Reply:
@skyemeedan Thanks. Models are not starting because mean-tokens has been removed from huggingface :cry: . I've opened a separate issue about this. Thankfully Live is fine as the model is cached

@ahmednasserswe Can you please take the above comments into account if you decide to use this PR in your work?